elfloader-tool/CMakeLists.txt

```
@@ -272,9 +272,9 @@ if(NOT "${IMAGE_START_ADDR}" STREQUAL "")
     add_custom_command(
         OUTPUT "${PLATFORM_INFO_H}" "${IMAGE_START_ADDR_H}"
         COMMAND
-            echo "#pragma once\\n\/* no platform YAML file available */" > "${PLATFORM_INFO_H}"
+            echo -e "#pragma once\\n\/* no platform YAML file available */" > "${PLATFORM_INFO_H}"
```

```suggestion
            echo -e "#pragma once\\n/* no platform YAML file available */" > "${PLATFORM_INFO_H}"
```
I think the extra backslash after the `\\n` is causing a failure now.

---

Nice to see that the tests are useful :-)

---

Different combinations of shells and echo programs have different behaviors.

This is what I currently have:
```
diff --git a/elfloader-tool/CMakeLists.txt b/elfloader-tool/CMakeLists.txt
index 8ab64d6..56e5a90 100644
--- a/elfloader-tool/CMakeLists.txt
+++ b/elfloader-tool/CMakeLists.txt
@@ -272,10 +272,9 @@ if(NOT "${IMAGE_START_ADDR}" STREQUAL "")
     add_custom_command(
         OUTPUT "${PLATFORM_INFO_H}" "${IMAGE_START_ADDR_H}"
         COMMAND
-            echo -e "#pragma once\\n\/* no platform YAML file available */" > "${PLATFORM_INFO_H}"
+            bash -c "echo -e \"#pragma once\\n\/* no platform YAML file available */\" > ${PLATFORM_INFO_H}"
         COMMAND
-            echo -e "#pragma once\\n#define IMAGE_START_ADDR ${IMAGE_START_ADDR}" >
-            "${IMAGE_START_ADDR_H}"
+            bash -c "echo -e \"#pragma once\\n#define IMAGE_START_ADDR ${IMAGE_START_ADDR}\" > ${IMAGE_START_ADDR_H}"
         VERBATIM
     )
 
```

---

This part of the scripts used to use file(Write). Is anyone against changing it back? (https://github.com/seL4/seL4_tools/pull/74#issuecomment-842856262)

---

The problem with with CMake's `file(WRITE ...)` was, that is cannot be used with `add_custom_command()`. So it has to be done once when CMake is initialized. The current solution can create this dynamically when rebuilding and changes are detected. I could not come up with a way to do this using `file(WRITE ...)` If echo makes problem, we could try `printf()`there. Or an inline python script ;)

---

The easiest way if `echo -e` has different behaviour on different shells is probably to use two separate `echo` commands for the two lines. The backslash at `\/*` is safe to remove in any case, though. If interpreted correctly it's just `/` anyway.

---

I tried to get it in one line back then to make it atomic.

---

> I tried to get it in one line back then to make it atomic.

It doesn't need to be atomic because the build step itself is atomic.

> The problem with with CMake's file(WRITE ...) was, that is cannot be used with add_custom_command()

Yes, but in the original code, the content being generated by file(WRITE) doesn't change dynamically. Changing the value of `IMAGE_START_ADDR` requires rerunning CMake which regenerates the file.